kolibri/plugins/coach/assets/src/modules/lessonResources/handlers.js

```diff
@@ -226,3 +200,20 @@ function _prepLessonContentPreview(store, classId, lessonId, contentId) {
     }
   );
 }
+
+export function showLessonResourceSearchPage(store, params) {
+  return store.dispatch('loading').then(() => {
+    return ContentNodeSearchResource.fetchCollection({
+      getParams: {
+        search: params.searchTerm,
+      },
+    }).then(results => {
```

Comment:
To do filtering that relies on the entire search set, just need to destructure assignment here:
`{ results, channel_ids, content_kinds, total_results }` instead of `results`.